express-notes-api/index.js

```diff
@@ -0,0 +1,199 @@
+// We want to be able to use express and this will allow us to
+const express = require('express');
+const appObj = express();
+// we want to be able to manipulate the file system with the fs module
+const fs = require('fs');
+// We want to be able to use data from the data.json file
+const dataJSON = require('./data.json');
+// this is the notes object where the notes are held
+const notesObj = dataJSON.notes;
+const dataId = dataJSON.nextId;
+
+// this will allow the server to listen for requests coming in
+appObj.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('The server is now listening.');
+});
+
+// this method will get the the data from the data.json object
+// it will place all the note objects into an array
+// then the server will send a json response of the array
+// if there are no notes then the response will be an empty array
+appObj.get('/api/notes', (req, res) => {
+  const gradeArr = [];
+  if (dataJSON === {}) {
+    res.json(gradeArr);
+  }
```

Comment:
You can leave this `if` statement out for a couple of reasons.

a) objects never equal each other, even if they have the same properties. two references to the _same_ object will be equal, but that's it, so `{} === {}` is `false`.

b) If your `dataJSON.notes` happens to be empty, the `for ... in` loop won't iterate at all, and you'll send back the empty stringified `gradeArr` all the same.